src/assets/scripts/client/changelog/ChangelogController.js

```diff
@@ -0,0 +1,223 @@
+         *
+         * @property version
+         * @type {String}
+         */
+        this.version = null;
+
+        /**
+         * A boolean flag, to determine whether or not
+         * we've loaded the changelog
+         *
+         * @property isLoaded
+         * @type {Boolean}
+         */
+        this.isLoaded = false;
+
+        this._setupHandlers()
+            .enable();
+    }
+
+    /**
+     * Sets up the event handlers
+     *
+     * @for ChangelogController
+     * @method _setupHandlers
+     * @private
+     * @chainable
+     */
+    _setupHandlers() {
+        this._onChangelogToggleHandler = this._onChangelogToggle.bind(this);
+
+        this.$changelogTrigger.on('click', this._onChangelogToggleHandler);
+        this.$changelogDismiss.on('click', this._onChangelogToggleHandler);
+
+        return this;
+    }
+
+    /**
+     * Sets the changelog data
+     *
+     * @for ChangelogController
+     * @method enable
+     */
+    enable() {
+        this.content = '<p>Loading...</p>';
+        this.version = window.GLOBAL.VERSION;
+
+        this.$changelogData.html(this.content);
+        this.$changelogContainer.append(this.$changelogData);
+        this.$changelogContainer.append(this.$changelogDismiss);
+        this.$element.append(this.$changelogContainer);
+
+        this.loadChangelogContent();
+    }
+
+    /**
+     * Toggles visibility.
+     *
+     * @for ChangelogController
+     * @method _onChangelogToggle
+     * @private
+     */
+    _onChangelogToggle() {
+        console.log('toggled changelog recieved')
+        this.$changelogContainer.toggleClass(SELECTORS.CLASSNAMES.CHANGELOG_VISIBLE);
+    }
+
+    /**
+     * Determines whether the user has played this version before,
+     * and if the changelog should display on load.
+     *
+     * @for ChangelogController
+     * @method _shouldToggleOnStart
+     * @returns {Boolean} if the user has not played this version
+     */
+    _shouldToggleOnStart() {
+        const lastPlayedVersion = localStorage[STORAGE_KEY.ATC_LAST_VERSION];
+        const currentVersion = this.version;
+        const shouldDisplayChangelog = lastPlayedVersion !== currentVersion;
+
+        if (shouldDisplayChangelog) {
+            localStorage[STORAGE_KEY.ATC_LAST_VERSION] = currentVersion;
+        }
+
+        return shouldDisplayChangelog;
+    }
+
+    /**
+     * Calls a changelog loader asynchronously. Calls `load_complete` when
+     * the changelog data is successfully loaded. Stores data in
+     * `this.content`.
+     *
+     * @for ChangelogController
+     * @method loadChangelogContent
+     */
+    loadChangelogContent() {
+        const options = {
+            url: 'assets/changelog.json',
+            immediate: true
+        };
+        const changelogPromise = this.contentQueue.add(options);
+
+        changelogPromise.done((data, textStatus, jqXHR) => {
+            this.content = data.changelog;
+            this.load_complete();
```

Comment:
No underscores in method names.  Also, does this need to be a blind handler?

Reply:
I believe it does need to be bound